### DISCUSSION TITLE
webdriver-io closeWindow is not triggering beforeunload event

### DISCUSSION THREAD

**Discussion by mikemcg52** [webdriverio/webdriverio#11013] [Technical Discussion]
I'm working on a Vue 3 project and need to track users opening and closing tabs.  Everything is working pretty well and I'm creating a set of webdriver.io tests to cover all of the use cases.  The issue I'm encountering is that the closeWindow command is not triggering the window.beforeunload event.  I've created a minimal reproduction of this that I posted on stackoverflow:

[https://stackoverflow.com/questions/76922670/webdriver-io-closewindow-is-not-triggering-beforeunload-event](url)

**Comment by christian-bromann** [comment]
Thanks for raising an issue @mikemcg52. Unfortunately this is not something we can fix within WebdriverIO and should be addressed on the protocol level. There is already an issue for this open: https://github.com/w3c/webdriver/issues/1294. I recommend to drop a comment in there with your feedback so the working group has a better idea what to do.

**Reply by mikemcg52** [reply]
@christian-bromann Thanks for your reply.  Now that I know the situation I can try to figure out a possible workaround in my tests.  I'll also drop by the w3c link you provided and leave a comment as you suggested.

**Reply by christian-bromann** [reply]
Awesome, thanks!
